TLM/TLM/Manager/Impl/AdvancedParkingManager.cs

```diff
@@ -2153,32 +2153,57 @@ private ExtSoftPathState
             return false;
         }
 
-        internal static void SwapParkedVehicleWithElectric(
+        /// <summary>
+        /// Swaps parked car with electric, returns electric vehicle info on success
+        /// </summary>
+        /// <param name="logParkingAi">should log</param>
+        /// <param name="citizenId">citizen id</param>
+        /// <param name="citizen">citizen data</param>
+        /// <param name="position">parked vehicle position</param>
+        /// <param name="rotation">parked vehicle rotation</param>
+        /// <param name="electricVehicleInfo">electric vehicle info</param>
+        /// <returns>true only if electric vehicle info exists and parked vehicle was swapped
+        /// otherwise false
+        /// </returns>
+        internal static bool SwapParkedVehicleWithElectric(
                                                     bool logParkingAi,
                                                     uint citizenId,
                                                     ref Citizen citizen,
                                                     Vector3 position,
                                                     Quaternion rotation,
```

Comment:
I'm not sure if it makes any difference here, but passing the `position` (3 x 4 byte) and `rotation` (4 x 4 byte) structs totals 28 bytes. I assume it creates defensive copies of them too?

Would it be better to just pass `ref` to the `VehicleParked` struct (total of 8 bytes)?

Reply:
Maybe but.. it's not running in a loop and there are tons of other more memory pressured places in vanilla or TM:PE. 

IMO it's just too small to bother -> creating ref and then dereference takes time so I'm not so sure if it's worth especially when usually vectors or quaternions are passed by ref if you expect to modify it inside the method. 
If struct would have like 10 fields, sure it would have much more sense to me to pass it by ref.

Reply:
Also that does not differ too much compared to number of variables you use in the method -> stack of specific size needs to be allocated each time you call the method. 🤷‍♂️